public/js/Player.js

```diff
@@ -14,15 +14,37 @@ class Player extends GameObject {
 	_score: number;
 	_shootDir: Array<number>;
 	_createBullet: Function;
+	_name: string;
+	_init_name: boolean;
 
-	constructor(startX: number, startY: number, dir: Array<number>, color: string, createBullet: Function) {
+	constructor(startX: number, startY: number, dir: Array<number>, color: string, name: string, createBullet: Function) {
 		super(startX, startY, Constants.playerSize, Constants.playerSize, color);
 		this._dir = dir;
 		this._shootDir = dir;
 		this._bulletCount = 1;
 		this._chargeTime = 0;
 		this._createBullet = createBullet;
 		this._score = 0;
+		this._init_name = false;
+		this._name = this._generateName(name);
+	}
+
+	_generateName(name: string) {
+		if (name != null) {
+			return name;
+		}
+
+		if (!this._init_name) {
```

Comment:
Hmm, why do you need __init__name? Can't you just check if this._name is null or something along those lines? Seems messy to have another bool for this